webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/InterpolatedValueCompiler.scala

```diff
@@ -172,6 +173,7 @@ object PropertyParserTokens {
   final case class Param(path: List[String])       extends AnyVal with Interpolation
   //here, we keep the case as it is given
   final case class Property(path: List[String], opt: Option[PropertyOption]) extends Interpolation
+  final case class RudderEngine(engine: String, method: List[String], param: Map[String, ConfigValue] ) extends Interpolation
```

Comment:
Not sure about the `param: Map[String, ConfigValue]`, can we only use `ConfigValue` ?

Reply:
yes, it shoudl be only config value

Reply:
Wouldn't it more robust if we create :
```scala
trait Engine {
   def name: String
   def compute: String //logic here to get value
 }
 ```
instead of `engine: String` in `RudderEngine`

(Not sure if it will be expandable for plugins with the `trait Engine`